Add more pieces to dev tools docs (#4322)
project-koku/koku: docs/devtools.md

# Developer Tools

This section describes tooling and features included in Koku to assist developers contributing to Koku.

## Environment Variables

Koku makes use of environment variables to configure features and application capabilities.

> In the repository, there is an `.env.example` file with sample environment settings. The `.env` file is used by Django\'s tools as well as Koku\'s scripting. It is the recommended way to configure your local development environment settings.

Reviewer: Escaping the single quotes here is craft from the RST conversion and can be removed.
```suggestion
In the repository, there is an `.env.example` file with sample environment settings. The `.env` file is used by Django's tools as well as Koku's scripting. It is the recommended way to configure your local development environment settings.
```

## PostgreSQL Tips:

**Connect to Postgre**
```
PGPASSWORD=postgres psql postgres -U postgres -h localhost -p 15432
```

> 

Reviewer: Another handy way of managing connections is to use a [connection service file](https://www.postgresql.org/docs/current/libpq-pgservice.html).

Example `~/.pg_service.conf`:
```ini
[koku_db]
host=localhost
dbname=postgres
port=5432
user=postgres
password=postgres
```

Then you can set the `PGSERVICE` variable and connect.
```
PGSERVICE=koku_db psql
```

Author: @samdoran I like this idea, you want to push a commit up to the branch suggesting this alternative too?

Author: Added!

```
python3 dev/scripts/render_templated_sql.py -f /path/to/template.sql -b 11
```

## Debugging Django ORM

> The following code snippit will provide you with queries generate by Django that are able to be copy & pasted into postgresql.

Reviewer: ```suggestion
The following code snippit will provide you with queries generate by Django that are able to be copy and pasted into postgresql.
```

```
from django.db import connection
readable_queries = []
for dikt in connection.queries:
    for key, item in dikt.items():
        item = item.replace("\"", "")
        item = item.replace("\\", "")
        readable_queries.append({key: item})
LOG.info(readable_queries)
```

If you are debugging queries generated for the report endpoints, a good place to put this code is in the `ReportView` class since it is the parent class for all the views.

## How to check Redis Keys

```

> docker exec -it koku_redis bash
> root@04b44144ce2a:/data# redis-cli

Reviewer: ```suggestion
docker exec -it koku_redis redis-cli
```
